File: old-projects/bd/bd.py
from collections import defaultdict
import random

DAYS = 365
results = [0 for i in range(DAYS)]
N = 100

for n in range(N):
    bdays = defaultdict(int)
    for i in range(1,DAYS):
        bdays[random.randint(1,DAYS)] += 1
        if len(bdays) != i:
            break
    for _i in range(i,DAYS):
        results[_i] += 1

for n in results[:50]:
    print int(n*100./N)

File: old-projects/bd/bd.js
document.write("bd");

function randint(hi,lo) { return Math.floor(Math.random()*(hi-lo) + lo) }

function plotBirthdays(n,days) {
    var results = [];
    for (var i=0;i<days;i++) { results.push(0); }
    for (var _run=0;_run<n;_run++) {
        var bdays = {};
        for (var people_in_room=0;people_in_room<days;people_in_room++) {
            var bday = randint(1,days);
            if (bdays[bday]) { break; }
            bdays[bday] = (bdays[bday] ||0) + 1;
        }
        for (var j_d=people_in_room;j_d<results.length;j_d++) {
            results[j_d] += 1;
        }
    }
    var percentages = results.map(function(count) { return Math.round(100*count/n); })
}

plotBirthdays(100,365);